sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2106,6 +2106,15 @@ object SQLConf {
       .booleanConf
       .createWithDefault(true)
 
+  val OPTIMIZER_IGNORE_HINTS =
```

Comment:
Shall we use `IGNORE_HINTS` instead of `OPTIMIZER_IGNORE_HINTS`? We don't need to use `OPTIMIZER_` prefix usually.

Reply:
@dongjoon-hyun Thanks.. so you would prefer to drop "optimizer" from the config name and call it "spark.sql.ignoreHints" instead ?

Reply:
How about `spark.sql.ignorePlanHints`?

Reply:
Thanks @maropu. Will make the change.

Reply:
No, @dilipbiswal . I meant only the variable name, not a conf name.
The conf name `spark.sql.optimizer.xxx` looked okay to me.

Reply:
Both names (`spark.sql.optimizer.ignoreHints` and `spark.sql.ignorePlanHints`) is okay to me.

Reply:
@maropu @dongjoon-hyun 
I have it as `spark.sql.ignorePlanHints` in my recent commit. Please let me know if its okay.. i am fine to change it back to `spark.sql.optimizer.ignoreHints` if required.

Reply:
Since it's a namespace, `spark.sql.optimizer` looks better than `spark.sql` because it's more specific. However, let's wait for @cloud-fan 's advice for `.enabled is required or not`.

Reply:
According to the config naming policy (in `core/src/main/scala/org/apache/spark/internal/config/ConfigEntry.scala`), `.enabled` is required if there are sub-configs. Otherwise it's optional. `spark.sql.optimizer.disableHints` is fine.

Reply:
@cloud-fan Thanks a lot.

Reply:
Thank you, @cloud-fan !